packages/flutter/test/material/ink_sparkle_test.dart

```diff
@@ -0,0 +1,150 @@
+          ),
+        ),
+      ),
+    ));
+    final Finder buttonFinder = find.text('Sparkle!');
+    await tester.tap(buttonFinder);
+    await tester.pump();
+    await tester.pumpAndSettle();
+
+    final MaterialInkController material = Material.of(tester.element(buttonFinder))!;
+    await tester.pump(const Duration(milliseconds: 200));
+    expect(material, paintsExactlyCountTimes(#drawPaint, 1));
+  },
+    skip: kIsWeb, // [intended] SPIR-V shaders are not yet supported for web.
+  );
+
+  /////////////
+  // Goldens //
+  /////////////
+
+  testWidgets('InkSparkle renders with sparkles when top left of button is tapped', (WidgetTester tester) async {
+    await _runTest(tester, 'top_left', 0.2);
+  },
+    skip: kIsWeb, // [intended] SPIR-V shaders are not yet supported for web.
+  );
+
+  testWidgets('InkSparkle renders with sparkles when center of button is tapped', (WidgetTester tester) async {
+    await _runTest(tester, 'center', 0.5);
+  },
+    skip: kIsWeb, // [intended] SPIR-V shaders are not yet supported for web.
+  );
+
+  testWidgets('InkSparkle renders with sparkles when bottom right of button is tapped', (WidgetTester tester) async {
+    await _runTest(tester, 'bottom_right', 0.8);
+  },
+    skip: kIsWeb, // [intended] SPIR-V shaders are not yet supported for web.
+  );
+}
+
+Future<void> _runTest(WidgetTester tester, String positionName, double distanceFromTopLeft) async {
+   final Key repaintKey = UniqueKey();
+   final Key buttonKey = UniqueKey();
+   await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: Center(
+          child: RepaintBoundary(
+            key: repaintKey,
+            child: ElevatedButton(
+              key: buttonKey,
+              style: ElevatedButton.styleFrom(splashFactory: InkSparkle.constantTurbulenceSeedSplashFactory),
+              child: const Text('Sparkle!'),
+              onPressed: () { },
+            ),
+          ),
+        ),
+      ),
+    ));
+    final Finder buttonFinder = find.byKey(buttonKey);
+    final Finder repaintFinder = find.byKey(repaintKey);
+
+    await _warmUpShader(tester, buttonFinder);
+
+    final Offset topLeft = tester.getTopLeft(buttonFinder);
+    final Offset bottomRight = tester.getBottomRight(buttonFinder);
+
+    final Offset target = topLeft + (bottomRight - topLeft) * distanceFromTopLeft;
+    await tester.tapAt(target);
+    for (int i = 0; i <= 5; i++) {
+      await tester.pump(const Duration(milliseconds: 50));
+      await expectLater(
+        repaintFinder,
+        matchesGoldenFile('ink_sparkle.$positionName.$i.png'),
+      );
+    }
+}
+
+// Warm up shader. Compilation is of the order of 10 milliseconds and
+// Animation is < 1000 milliseconds. Use 2000 milliseconds as a safety
+// net to prevent flakiness.
+Future<void> _warmUpShader(WidgetTester tester, Finder buttonFinder) async {
+  await tester.tap(buttonFinder);
+  await tester.pumpAndSettle(const Duration(milliseconds: 2000));
```

Comment:
Is there a way to avoid using a specific duration?

Reply:
In theory pumpAndSettle should be enough, but wanted to add a safety net to avoid possible flakiness